### PR TITLE
Task/remove unused files

### PR DESCRIPTION
**Pull request by mikemhenry**
## Motivation and Context

Resolves: #716 

## Change log

```

```

## Checklist:

- [ ] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/master/CONTRIBUTING.md).
- [ ] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/master/ContributorAgreement.md).
- [ ] My name is on the [list of contributors](https://github.com/glotzerlab/hoomd-blue/blob/master/sphinx-doc/credits.rst).

### CODE REVIEW

**Comment by mikemhenry** [comment]
Okay I think I need to wait for #735 and #734 to get merged in before I'm able to really test this PR since I removed `hoomd.context` and `hoomd.group` but not completely, so I'm getting import errors...

File: README.md [hunk 1]
```diff
@@ -44,27 +44,6 @@ workflows all with Python code in your job.
 
 Here is a simple example:
```
Comment (joaander) [inline README.md:45]: We either need to replace this with a functioning example or remove the whole "Here is a simple example" section including the header.

File: hoomd/md/constrain.py [hunk 2]
```diff
@@ -281,31 +268,6 @@ class rigid(_ConstraintForce):
 
     Example that creates rigid rods::
```
Comment (joaander) [inline hoomd/md/constrain.py:269]: Remove the heading too, since you removed the entire section.

**Comment by mikemhenry** [comment]
I will merge the branch back in and fix those sections later this afternoon. I think for now I will remove the headers and I'll open two issues so we can remember to add back in examples.

**Comment by mikemhenry** [comment]
Oh boy not sure what went wrong with my merge but I managed to rebase on top of my branch so its 🌈 double commit  all the way🌈... good thing we squash!

**Comment by mikemhenry** [comment]
I put the PR back into draft mode, I don't want to tax the CI and I haven't run tests locally yet.

**Comment by mikemhenry** [comment]
Okay I have done something to really mess up the git history somehow, at least on the web interface since locally: `git diff upstream/feature/new-object-API .azp/templates/run_tests.yml` produces no output, but the diff on github shows some things added.

**Comment by joaander** [comment]
You haven't done anything wrong. I don't know if it is a feature or a bug, but there are commits on `feature/new-object-API` that are not on your branch. Those changes appear in your PR diff even though they will not be applied. To fix, merge `feature/new-object-API` into your branch and then github will only show your changes.

**Comment by mikemhenry** [comment]
Okay merged in the changes, hopefully everything passes!

**Review by joaander — APPROVED** [review]
Great! Thanks for the clean up- especially that 500k line log output file from one of the tests.

Do you have bandwidth available to work on additional v3 issues or PR review?

**Pull request merged**